src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,570 @@
+            model=self._model, frame_name=right_foot_sole_frame_name
+        )
+
+        # Check that joint list from mujoco and jaxsim have the same elements (just ordered differently)
+        assert np.array_equal(
+            np.array(js_joint_names)[self._to_user], user_joint_names
+        ), "Joint names mismatch"
+        assert np.array_equal(
+            np.array(user_joint_names)[self._to_js], js_joint_names
+        ), "Joint names mismatch"
+
+        # Configure visualization
+
+        self._visualization_mode = visualization_mode
+
+        if self._visualization_mode is not None:
+            if self._visualization_mode not in ["record", "interactive"]:
+                raise ValueError(
+                    f"Invalid visualization mode: {self._visualization_mode}. "
+                    f"Valid options are: 'record', 'interactive'"
+                )
+
+            mjcf_string, assets = UrdfToMjcf.convert(
+                urdf=self._model.built_from,
+            )
+
+            self._mj_model_helper = MujocoModelHelper.build_from_xml(
+                mjcf_description=mjcf_string, assets=assets
+            )
+
+            self._recorder = MujocoVideoRecorder(
+                model=self._mj_model_helper.model,
+                data=self._mj_model_helper.data,
+                fps=30,
+            )
+
+        self._is_initialized = True
+
+    def set_input(self, input: npt.ArrayLike) -> None:
+        """Set the robot input torques.
+
+        Args:
+            input: torques to apply to the robot joints.
+        """
+
+        self._tau = jnp.array(input)[self._to_js]
+
+    def step(self, n_step: int = 1, dry_run=False) -> None:
+        """Step the simulation forward by n_step steps.
+
+        Args:
+            n_step: number of steps to simulate.
+            dry_run: If True, the simulation will not advance the state of the
+                simulator and will not trigger any visualization.
+        """
+
+        assert (
+            self._is_initialized
+        ), "Simulator is not initialized, call load_model first."
+
+        for _ in range(n_step):
+            if self._contact_model_type is JaxsimContactModelEnum.VISCO_ELASTIC:
+                self._data, _ = jaxsim.rbda.contacts.visco_elastic.step(
+                    model=self._model,
+                    data=self._data,
+                    dt=self._dt,
+                    joint_force_references=self._tau,
+                    link_forces=None,
+                )
+
+            else:
+                # All other contact models
+                self._data, self._integrator_state = js.model.step(
+                    model=self._model,
+                    data=self._data,
+                    dt=self._dt,
+                    integrator=self._integrator,
+                    integrator_state=self._integrator_state,
+                    joint_force_references=self._tau,
+                    link_forces=None,
+                )
```

Comment:
@xela-95 can you please try if this works in this updated branch?

```suggestion
                # All other contact models
                with self._data.switch_velocity_representation(VelRepr.Inertial):
                    self._data, self._integrator_state = js.model.step(
                        model=self._model,
                        data=self._data,
                        dt=self._dt,
                        integrator=self._integrator,
                        integrator_state=self._integrator_state,
                        joint_force_references=self._tau,
                        link_forces=None,
                    )
```

Reply:
@diegoferigo I performed a simulation with the modifications you suggested here and the result is good, with stickbot walking. Here's the rendering: 

https://github.com/user-attachments/assets/413fbc9b-f594-4e46-a69a-de9a43f69a43

Reply:
Great, thanks for the test. I'll try to rebase my code then, probably the old branch I'm using (from your early experiments) is out of sync and there missing improvements.

Reply:
Ok let me know if you need additional tests!